server/graphql/v1/mutations/orders.js

```diff
@@ -569,8 +570,7 @@ export async function updateSubscription(remoteUser, args) {
     });
 
     const newSubscription = await models.Subscription.create(newSubscriptionDataValues);
-
-    const newOrderDataValues = Object.assign(omit(order.dataValues, ['id']), {
+    const newOrderDataValues = Object.assign(omit(order.dataValues, ['id', 'status']), {
```

Comment:
I did it and removed because I thought the orders would always be `ACTIVE` but they are usually `PENDING` instead and that's the default status. I believe it makes sense because this new Order wasn't executed yet and It becomes active after the first execution.
Though i could make this new order with the same status of the "original" one, I still think that in my opinion it should be PENDING.
Given that, what do you think works best?

Reply:
@marcogbarcellos where is that updated from `PENDING` to `ACTIVE`?

Reply:
@znarf it only changes the status from `PENDING` to `ACTIVE` when waiting the payment methods `processOrder`( a more concrete case would be waiting on stripe to confirm payment and then right away update the order again from PENDING to ACTIVE). but in this PR's case for the orders to change from `PENDING` to `ACTIVE` I needed to do one more change. Look at https://github.com/opencollective/opencollective-api/blob/753d48825039b182b27077ecbe804d5b4fca678f/server/lib/subscriptions.js#L72-L97

I could still "reuse" the original Order which is usually `ACTIVE`(my bad for the comment above, I assumed it wrongly because the orders have a `PENDING` status on the test file, and we execute both `PENDING` and `ACTIVE` orders regardless) but I really think it's better keep it PENDING and only change on the first execution.

a use case would be:
- You donate 2USD **monthly** to mochajs in November 15th, thus creates an Order and its status is `ACTIVE`.
- you update that subscription to donate 10USD in November 23rd. this will, on the background, deactivate the original subscription and create a new one as well as generate a new Order.
- This new Order would have the status `PENDING` because it'll only be executed in December 1st. And it'll become active as soon as it's executed for the first time(transactions created).

PS: To check **Order** default status, see : https://github.com/opencollective/opencollective-api/blob/master/migrations/20180727173834-add-status-to-order.js

Reply:
@marcogbarcellos Thank you for the explanations! Sounds good.

Can we then make it explicitly `PENDING` in the code we're discussing?

Reply:
done.

Reply:
@marcogbarcellos I don't see it, where?